pkg/azuredisk/azuredisk_v2.go

```diff
@@ -405,11 +408,46 @@ func (d *DriverV2) StartControllersAndDieOnExit(ctx context.Context) {
 		klog.Errorf("Failed to initialize NodeAvailabilityController. Error: %v. Exiting application...", err)
 		os.Exit(1)
 	}
-	err = controller.WatchObject(mgr, source.Kind{Type: &v1.Namespace{}})
+	err = controller.WatchObject(mgr, source.Kind{Type: &v1.Namespace{}}, predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool {
```

Comment:
@sunpa93 seems like we are creating a noop controller here. What is the point of passing the predicates in this case? Even if the predicate returns true, nothing will be triggered.

Reply:
It might also be good to rename this `CacheObject` if that is the intend of creating the noop controller.

Reply:
> @sunpa93 seems like we are creating a noop controller here. What is the point of passing the predicates in this case? Even if the predicate returns true, nothing will be triggered.

The reason we create a noop controller is so that we can use the same cached client to fetch namespaces instead of creating a dedicated informer for it to store in the shared state, which will require a separate syntax for fetching objects.

Reply:
Unfortunately, the pkg doesn't seem to offer us a way to inject our own informer to its cached reader.

Reply:
Yeah, I understand that part, but my question here is - do we need the predicates? We shouldn't need then for a noop controller. Also, we should probably rename the function like suggested above for more clarity.

Reply:
Without the predicates returning false, we would have the namespace objects get added to the internal queue for reconcile operation, which will return with a noop. So, I figured it would be a waste of memory.

Reply:
I agree with the naming. I think Huichan has already changed the name of the functions.

Reply:
I renamed the functions to `CacheNamespace` and `CacheCRD`.

Reply:
What I was saying is that all predicates should return false, because there is no controller to reconcile even if they return true

Reply:
As I left in the earlier comments of the PR, the predicates for these noop reconcilers should return false.